axelrod/moran.py

```diff
@@ -31,7 +31,33 @@ def fitness_proportionate_selection(scores):
 
 
 class MoranProcess(object):
-    def __init__(self, players, turns=100, noise=0, deterministic_cache=None):
+    def __init__(self, players, turns=100, noise=0, deterministic_cache=None, mutation_rate=0.):
+        """
+        An agent based Moran process class. In each round, each player plays a Match with each other
+        player. Players are assigned a fitness score by their total score from all matches in the round.
+        A player is chosen to reproduce proportionally to fitness, possibly mutated, and is cloned. The
+        clone replaces a randomly chosen player.
+
+        If the mutation_rate is 0, the population will eventually fixate on exactly one player type. In this
+        case a StopIteration exception is raised and the play stops. If mutation_rate is not zero, then
+        the process will iterate indefinitely, so mp.play() will never exit, and you should use the class as an
+        iterator instead.
+
+        When a player mutates it chooses a random player type from the initial population. This is not the only
+        method yet emulates the common method in the literature.
+
+        Parameters
+        ----------
+        players, iterable of axelrod.Player subclasses
+        turns: int, 100
+            The number of turns in each pairwise interaction
+        noise: float, 0
+            The background noise, if any. Randomly flips plays with probability `noise`.
+        deterministic_cache: axelrod.DeterministicCache, None
+            A optional prebuilt deterministic cache
+        mutation_rate: float, 0
```

Comment:
A lot of the literature I've seen write this in a generic way with a matrix $M$ for mutation where $M_ij$ is the probability if mutating from species $i$ to $j$. Is that worth doing (perhaps at a later date)?

More of a question than a request. We could simply have that the mutation rate could either be a float or a 2 d array... (Or whatever: not suggest this needs to be done here).